### ISSUE TITLE
False positives for custom property sets followed by a comment in no-extra-semicolons

### ISSUE THREAD

**Issue by alienlebarge** [stylelint/stylelint#2395]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

bug

> Which rule, if any, is this issue related to?

`no-extra-semicolons`

> What CSS is needed to reproduce this issue?

```css
/* ========================================================================
   #TYPOGRAPHY
   ======================================================================== */

:root {
    /**
     * Mor readable bodytext
     * see http://practice.typekit.com/lesson/caring-about-opentype-features/
     */
    --font-bodytext: {
        font-variant-ligatures: common-ligatures, contextual;
        font-variant-numeric: oldstyle-nums, proportional-nums;
        font-feature-settings: "kern", "liga", "clig", "calt", "onum", "pnum";
        font-kerning: normal;
    }; /* <--- get an error */





    /**
     * Mor readable headings
     * see http://practice.typekit.com/lesson/caring-about-opentype-features/
     */
    --font-headings: {
        font-variant-ligatures: common-ligatures, contextual,
            discretionary-ligatures;
        font-feature-settings: "kern", "liga", "clig", "calt", "dlig";
        font-kerning: normal;
    }; /* <--- get an error */





    /**
     * More readable sub text
     * see http://practice.typekit.com/lesson/caring-about-opentype-features/
     */
    --font-sub: {
        font-variant-ligatures: common-ligatures, contextual;
        font-variant-position: sub;
        font-feature-settings: "kern", "liga", "clig", "calt", "subs";
        font-kerning: normal;
    }; /* <--- get an error */





    /**
     * OpenType feature for small caps
     * see http://practice.typekit.com/lesson/caring-about-opentype-features/
     *
     * 1. Safari iOs doesn't support `font-feature` so we use `text-transform`
     *    to make the texte uppercase. http://caniuse.com/#search=font-feature
     */
    --font-small-caps: {
        font-variant-ligatures: common-ligatures, contextual;
        font-variant-numeric: oldstyle-nums, proportional-nums;
        font-variant-caps: all-small-caps;
        font-feature-settings: "kern", "liga", "clig", "calt", "onum", "pnum",
            "smcp", "c2sc";
        font-kerning: normal;
        text-transform: uppercase; /* [1] */
    }; /* <--- get an error */





    /**
     * OpenType feature for data
     * see http://practice.typekit.com/lesson/caring-about-opentype-features/
     */
    --font-data: {
        font-variant-ligatures: common-ligatures, contextual;
        font-variant-numeric: lining-nums, tabular-nums;
        font-feature-settings: "kern", "liga", "clig", "calt", "lnum", "tnum";
        font-kerning: normal;
    };
}

```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
    "rules": {
        "no-extra-semicolons": true
    }
}
```

> Which version of stylelint are you using?

e.g. `7.9.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint style.css`
and via *gulp-stylelint* in my projects

Here's a [gist](https://gist.github.com/alienlebarge/460cb9147ac2f6c0717215155a620f19) that reproduce the problem

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

It's related to Tabatkins [apply rule specification](https://tabatkins.github.io/specs/css-apply-rule/)

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. "The following warnings were flagged:"

```shell
style.css
 15:6  ✖  Unexpected extra semicolon   no-extra-semicolons
 30:6  ✖  Unexpected extra semicolon   no-extra-semicolons
 45:6  ✖  Unexpected extra semicolon   no-extra-semicolons
 66:6  ✖  Unexpected extra semicolon   no-extra-semicolons
```

**Comment by hudochenkov** [comment]
CSS `@apply` rule is “a Collection of Interesting Ideas.” I don't think stylelint will process correctly until it converted in spec. I suggest you to use [`stylelint-disable`](https://stylelint.io/user-guide/configuration/#turning-rules-off-from-within-your-css) comments to disable this rule:

```css
/* stylelint-disable no-extra-semicolons */
:root {
    /**
     * Mor readable bodytext
     * see http://practice.typekit.com/lesson/caring-about-opentype-features/
     */
    --font-bodytext: {
        font-variant-ligatures: common-ligatures, contextual;
        font-variant-numeric: oldstyle-nums, proportional-nums;
        font-feature-settings: "kern", "liga", "clig", "calt", "onum", "pnum";
        font-kerning: normal;
    };
}
/* stylelint-enable */
```

**Comment by alienlebarge** [comment]
So the following text in the [changelog v7.4.0](CHANGELOG.md#740) is a little bit confusing:
> Fixed: `no-extra-semicolons` now ignores the semicolon that comes after the closing brace of a custom property set.

**Comment by hudochenkov** [comment]
Thank you for pointing out! I was wrong. Stylelint do support custom property set for this rule.

I found out, that warning triggered if there a comment after custom property set. If these comments removed — no warnings triggered.

**Comment by davidtheclark** [comment]
Closed by #2396.
